### ISSUE TITLE
[Bug]: [DatePicker] [A11y] NVDA navigates just to "clickable" in browse mode

### ISSUE THREAD

**Issue by kolaps33** [microsoft/fluentui#30120]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 10 10.0.22631
    CPU: (12) x64 Intel(R) Xeon(R) E-2176M  CPU @ 2.70GHz
    Memory: 16.11 GB / 31.74 GB
  Browsers:
    Edge: Chromium (120.0.2210.77)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codesandbox.io/p/sandbox/kind-sun-rhf79d

### Bug Description

1. open url https://codesandbox.io/p/sandbox/kind-sun-rhf79d
2. run NVDA screen reader
3. set focus on "before" button 
4. press down arrow key to navigate until you reach "after" button
5. then press up arrow key to navigate until you reach "before" button
6. observe NVDA narraton

## Actual Behavior
NVDA navigate to element which narrates only "clickable"
Please see enclosed screenshot.
![image](https://github.com/microsoft/fluentui/assets/22620150/e1634657-5623-4171-a472-fb5a2ad6d777)

## Expected Behavior
there should be element with accessibility name which is narrated only "clickable"

## How to fix it
add `aria-hidden=true` on the span which is wrapping SVG icon
![image](https://github.com/microsoft/fluentui/assets/22620150/81a0dd65-e3be-49c1-af81-bb9616d80467)

### Requested priority

Normal

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
